### ISSUE TITLE
Error when I try to model P-norm using PowCone3d

### ISSUE THREAD

**Issue by dcajasn** [cvxpy/cvxpy#1528]
**Describe the bug**
I want to minimize the P-norm using the power cone based on the [MOSEK documentation](https://docs.mosek.com/modeling-cookbook/powo.html#sets-representable-using-the-power-cone).

**To Reproduce**
I got an error when I run following code:

import cvxpy as cp

T, N = 500, 5
d = cp.Variable((T, 1))
w = cp.Variable((N, 1))
t = cp.Variable()

rs = np.random.RandomState(123)
D = rs.randn(T, N)
ones = np.ones((T, 1))

constraints = [cp.sum(w) == 1,
               w >= 0,
               cp.constraints.PowCone3D(d, t * ones, D @ w, 0.5)]

risk = t
prob = cp.Problem(cp.Minimize(risk), constraints)
prob.solve(solver=cp.MOSEK)
weights = pd.DataFrame(w.value)
display(weights.T)

**Expected behavior**
Calculate the variable w, that minimize the P-norm of the product D @ w.

**Output**
The error message is:

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/opt/anaconda3/lib/python3.8/site-packages/mosek/__init__.py in appendconesseq(self, _arg2_ct, _arg3_conepar, _arg4_nummem, _arg5_j)
   1745         try:
-> 1746           _arg3_conepar_ = memoryview(_arg3_conepar)
   1747         except TypeError:

TypeError: memoryview: a bytes-like object is required, not 'list'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
~/opt/anaconda3/lib/python3.8/site-packages/mosek/__init__.py in appendconesseq(self, _arg2_ct, _arg3_conepar, _arg4_nummem, _arg5_j)
   1748           try:
-> 1749             _tmparr__arg3_conepar = array.array("d",_arg3_conepar)
   1750           except TypeError:

TypeError: must be real number, not list

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
/var/folders/2z/30r_gtpd1k9b7mlcdwftfthh0000gn/T/ipykernel_817/199135939.py in <module>
     16 risk = t
     17 prob = cp.Problem(cp.Minimize(risk), constraints)
---> 18 prob.solve(solver=cp.MOSEK)
     19 weights = pd.DataFrame(w.value)
     20 display(weights.T)

~/opt/anaconda3/lib/python3.8/site-packages/cvxpy/problems/problem.py in solve(self, *args, **kwargs)
    460         else:
    461             solve_func = Problem._solve
--> 462         return solve_func(self, *args, **kwargs)
    463 
    464     @classmethod

~/opt/anaconda3/lib/python3.8/site-packages/cvxpy/problems/problem.py in _solve(self, solver, warm_start, verbose, gp, qcp, requires_grad, enforce_dpp, **kwargs)
    956                     solving_chain.reductions[-1].name())
    957         start = time.time()
--> 958         solution = solving_chain.solve_via_data(
    959             self, data, warm_start, verbose, kwargs)
    960         end = time.time()

~/opt/anaconda3/lib/python3.8/site-packages/cvxpy/reductions/solvers/solving_chain.py in solve_via_data(self, problem, data, warm_start, verbose, solver_opts)
    341             a Solution object.
    342         """
--> 343         return self.solver.solve_via_data(data, warm_start, verbose,
    344                                           solver_opts, problem._solver_cache)

~/opt/anaconda3/lib/python3.8/site-packages/cvxpy/reductions/solvers/conic_solvers/mosek_conif.py in solve_via_data(self, data, warm_start, verbose, solver_opts, solver_cache)
    219                 env = mosek.Env()
    220                 task = env.Task(0, 0)
--> 221                 task = MOSEK._build_dualized_task(task, data)
    222         else:
    223             if len(data[s.C]) == 0:

~/opt/anaconda3/lib/python3.8/site-packages/cvxpy/reductions/solvers/conic_solvers/mosek_conif.py in _build_dualized_task(task, data)
    297         if num_dpow > 0:
    298             cones = [mosek.conetype.dpow] * num_dpow
--> 299             task.appendconesseq(cones, K[a2d.DUAL_POW3D], [3] * num_dpow, idx)
    300             idx += 3 * num_dpow
    301         num_psd = len(K[a2d.PSD])

~/opt/anaconda3/lib/python3.8/site-packages/mosek/__init__.py in appendconesseq(self, _arg2_ct, _arg3_conepar, _arg4_nummem, _arg5_j)
   1749             _tmparr__arg3_conepar = array.array("d",_arg3_conepar)
   1750           except TypeError:
-> 1751             raise TypeError("Argument _arg3_conepar has wrong type")
   1752           else:
   1753             _arg3_conepar_ = memoryview(_tmparr__arg3_conepar)

TypeError: Argument _arg3_conepar has wrong type

**Version**
 - OS: Mac OS Catalina 10.15.7
 - CVXPY Version: 1.1.15

**Additional context**
None

**Comment by SteveDiamond** [comment]
I looked into this, and the issue IMO is that ``K[a2d.DUAL_POW3D]`` in mosek_conif.py line 299 needs to be flattened (currently it's a list of lists). CC @rileyjmurray

**Comment by rileyjmurray** [comment]
Commit https://github.com/cvxpy/cvxpy/commit/3d29b98001f4750e744715e8c93c4c2eb88715b3 fixed this.

@SteveDiamond you were correct that the list needed to be flattened, but I dug deeper into why that was needed here when it wasn't needed elsewhere. The issue came down to how 2d inputs to ``PowCone3D`` and ``ExpCone`` are flattened to 1d representations. I modified ``cone_matrix_stuffing`` so that those constraints are always flattened before getting passed to a solver (analogous to how SOC constraints are always switched to have the axis oriented in a certain way). I've added ``pcp_3`` to ``solver_test_helpers`` to test this for MOSEK and SCS.

A minor note to @dcajasn: your code is missing a necessary constraint ``cp.sum(d) == t``.
